src/Endatix.Core/UseCases/FormDefinitions/PartialUpdateActive/PartialUpdateActiveFormDefinitionHandler.cs

```diff
@@ -18,10 +19,24 @@ public async Task<Result<FormDefinition>> Handle(PartialUpdateActiveFormDefiniti
             return Result.NotFound("Active form definition not found.");
         }
 
-        activeDefinition.UpdateSchema(request.JsonData);
-        activeDefinition.UpdateDraftStatus(request.IsDraft);
+        if(await FormDefinitionHasSubmissions(activeDefinition.Id) && activeDefinition.JsonData != request.JsonData) {
```

Comment:
I would recommend to reverse the if statements with the cheapest first and then go to the database. It's good practice to add the most expensive check (e.g. DB round trip) at the end and only check it when other if conditions lead to it.

Also, I would recommend extract this if check into a private method for readability e.g. `Task<bool> ShouldCreateNewFormDefinitionAsync()` will make the code easier to debug and read.